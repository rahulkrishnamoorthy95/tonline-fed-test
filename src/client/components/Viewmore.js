import React from 'react';

function Viewmore({ callback, hidden }) {

    if (hidden) {
        return null;
    }
    
    return (
        <div className="view-more">
            <button type="button" className="button" onClick={callback}>{'View More'}</button>
        </div>
    );
}

export default Viewmore;