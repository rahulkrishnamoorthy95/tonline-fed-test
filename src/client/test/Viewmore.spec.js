import React from 'react';
import Viewmore from '../components/Viewmore';

describe('Viewmore', () => {

	it("should render initial layout", () => {
        const component = shallow(<Viewmore />);
        expect(component.getElements()).toMatchSnapshot();
    });
    
    it('should render based on hidden prop', () => {
        const props = {
            hidden: true
        };
        let wrapper = mount(<Viewmore {...props} />);    
        expect(wrapper.prop('hidden')).toEqual(true);

        props.hidden = false;
        wrapper = mount(<Viewmore {...props} />);    
        expect(wrapper.prop('hidden')).toEqual(false);
      });

      it('Test button click event', () => {
        const mockCallBack = jest.fn();
        const button = shallow(<Viewmore callback={mockCallBack} />);
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toBe(1);
      });

});