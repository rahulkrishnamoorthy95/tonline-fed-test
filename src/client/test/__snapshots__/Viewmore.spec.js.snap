// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viewmore should render initial layout 1`] = `
Array [
  <div
    className="view-more"
  >
    <button
      className="button"
      type="button"
    >
      View More
    </button>
  </div>,
]
`;
